robotoff/insights/importer.py

```diff
@@ -847,24 +846,17 @@ def import_product_predictions(
         .tuples()
     )
 
-    to_import = itertools.chain.from_iterable(
-        (
-            (
-                create_prediction_model(
-                    prediction, product_predictions, server_domain, timestamp
-                )
-                for prediction in product_predictions.predictions
-                if (
-                    product_predictions.type,
-                    server_domain,
-                    product_predictions.source_image,
-                    prediction.value_tag,
-                    prediction.value,
-                )
-                not in existing_predictions
-            )
-            for product_predictions in product_predictions_iter
+    to_import = (
+        create_prediction_model(prediction, server_domain, timestamp)
+        for prediction in product_predictions_iter
+        if (
+            prediction.type,
+            server_domain,
+            prediction.source_image,
+            prediction.value_tag,
+            prediction.value,
         )
```

Comment:
maybe we could have a prediction.key() method ?

Reply:
I find it clearer this way, as we know which fields are used to consider a prediction as unique